dvc/data_cloud.py

```diff
@@ -74,7 +75,11 @@ def get_remote(self, remote=None, command="<command>"):
         )
 
     def _init_remote(self, remote):
-        return Remote(self.repo, name=remote)
+        try:
+            return self._remotes[remote]
+        except KeyError:
+            result = self._remotes[remote] = Remote(self.repo, name=remote)
```

Comment:
Could you please clarify why this is needed?

Reply:
1. For optimization, to not recreate remote and it's connections every time.
2.  To test AzureCompat. It doesn't have path in it, so it clashes with previous runs and fails.

Reply:
@Suor connections are not created when a remote is created. Connections are created when they are needed.

Reply:
@Suor Indeed, AzureCompat is using container's root. But it is a one single test, it shouldn't clash with others, since other azure tests use prefix within the container, so you might be covering up a bug introduced by your patch.